test/metabase/parameters/params_test.clj

```diff
@@ -118,7 +119,66 @@
+                                                                          ;; p3 - explicit :stage-number, id-based ref
+                                                                          {:card_id (u/the-id card)
+                                                                           :parameter_id "p3"
+                                                                           :target  [:dimension
+                                                                                     [:field (mt/id :products :id) nil]
+                                                                                     {:stage-number 0}]}
+                                                                          ;; p4 - explicit :stage-number, name-based ref
+                                                                          {:card_id (u/the-id card)
+                                                                           :parameter_id "p4"
+                                                                           :target  [:dimension
+                                                                                     [:field "ID" {:base-type :type/BigInteger}]
+                                                                                     {:stage-number 0}]}
+                                                                          ;; p5 - explicit :stage-number, post-aggregation stage
+                                                                          {:card_id (u/the-id card)
+                                                                           :parameter_id "p5"
+                                                                           :target  [:dimension
+                                                                                     [:field "CATEGORY" {:base-type :type/Text}]
+                                                                                     {:stage-number 1}]}
+                                                                          ;; p6 - explicit :stage-number, not existing stage
+                                                                          {:card_id (u/the-id card)
+                                                                           :parameter_id "p6"
+                                                                           :target  [:dimension
+                                                                                     [:field "CATEGORY" {:base-type :type/Text}]
+                                                                                     {:stage-number 2}]}]}]
+        (is (=? {"p1" [{:id (mt/id :products :id)}]
+                 "p2" [{:id (mt/id :products :id)}]
+                 "p3" [{:id (mt/id :products :id)}]
+                 "p4" [{:id (mt/id :products :id)}]
+                 "p5" [{:id (mt/id :products :category)}]}
```

Comment:
This won't fail if `"p6"` gets returned.

Reply:
Fixed